test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py

```diff
@@ -30,6 +31,11 @@
 GET_TEST_INFO_ARGUMENT = "--get_test_info"
 TEST_INFO_JSON_FILENAME = "test_info.json"
 
+# Pattern to match TC_*_number_number.py format
+# TC_ followed by cluster name (each part starts with letter),
+# then exactly _digit_digit.py
+TC_FILENAME_PATTERN = r"^TC_[A-Za-z][A-Za-z0-9]*(?:_[A-Za-z][A-Za-z0-9]*)*_\d+_\d+\.py$"
```

Comment:
Does the TC pattern accepts all of that?

I believe the following would pass in that regex:
`TC_Abcd123qwe_D987cba_1_2.py`

That said:
- Does the second part should accept alphanumeric or just letters?
- Does the letters should accept lowercase?
- The optional part inside `()` seems strange? Is that expected? Do we have examples like that?

My suggestion would be **something like**:
```suggestion
TC_FILENAME_PATTERN = r"^TC_[A-Za-z]{2,6}(_\d+){2}\.py$"
```

Or if lowercase letters are **not** acceptable use the following:

```suggestion
TC_FILENAME_PATTERN = r"^TC_[A-Z]{2,6}(_\d+){2}\.py$"
```